### PR TITLE
Repathed lockets, rings onto /jewelry.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Splits insignia, lockets and rings off accessory onto their own subtype.

## Why and what will this PR improve
Breaks #3798 up for ease of review.
Working towards destroying /obj/item/clothing/accessory.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Tested, working.

File: code/modules/clothing/jewelry/lockets.dm [hunk 1]
```diff
@@ -1,14 +1,12 @@
-/obj/item/clothing/accessory/locket
+/obj/item/clothing/jewelry/locket
```
Comment (out-of-phaze) [inline code/modules/clothing/jewelry/lockets.dm]: Seems a bit weird to me that a locket isn't a type of necklace, but I suppose that's fine. Not blocking. left this comment before I found a blocker.

File: tools/map_migrations/3902_jewelry.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/obj/item/clothing/accessory/locket/@SUBTYPES : /obj/item/clothing/jewelry/locket/@SUBTYPES{@OLD}
+/obj/item/clothing/accessory/chaplaininsignia/@SUBTYPES : /obj/item/clothing/jewelry/insignia/@SUBTYPES{@OLD}
+/obj/item/clothing/ring/@SUBTYPES : /obj/item/clothing/jewelry/ring/@SUBTYPES{@OLD}
+/obj/item/clothing/accessory/bracelet/@SUBTYPES : /obj/item/clothing/jewelry/bracelet/@SUBTYPES{@OLD}
+/obj/item/clothing/accessory/badge/tags/@SUBTYPES : /obj/item/clothing/jewelry/dog_tags/@SUBTYPES{@OLD}
```
Comment (out-of-phaze) [inline tools/map_migrations/3902_jewelry.txt]:
```suggestion
/obj/item/clothing/accessory/badge/tags/@SUBTYPES : /obj/item/clothing/jewelry/dog_tags/@SUBTYPES{@OLD}
/obj/item/clothing/accessory/necklace/@SUBTYPES : /obj/item/clothing/jewelry/necklace/@SUBTYPES{@OLD}
```

**Review by out-of-phaze — DISMISSED** [review]
A little wary of dog tags being made jewelry, but since the jewelry type doesn't actually show up in checks or have any functionality yet, I guess it's fine.

Aside from that, necklaces don't have a map migration. Whoops.

**Comment by MistakeNot4892** [comment]
Accidentally mashed in some changes that I need to split out again.

**Comment by MistakeNot4892** [comment]
Depends on #3931

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**